### ISSUE TITLE
Not tracking all the state changes evenif integrated the same way mentioned in the document. 

### ISSUE THREAD

**Issue by codalprashant** [revolunet/angular-google-analytics#219]
It is just tracking the first page. But when the state is changed, it stops tracking the pages and the active user.

**Comment by Sparksx** [comment]
Hi,

This is due to ui-router update, check this: https://ui-router.github.io/ng1/docs/latest/modules/ng1_state_events.html

**Comment by codalprashant** [comment]
I managed it using AnalyticsProvider.setPageEvent('$stateChangeStart') in config and $rootScope.$on('$stateChangeStart',  
      function(event, toState) { 
        Analytics.trackPage($location.absUrl());
}); in run section.

**Comment by revolunet** [comment]
so closable ?

**Comment by Sparksx** [comment]
@revolunet No! You need to fix it in your code

`AnalyticsProvider.setPageEvent('$stateChangeSuccess');` is now deprecated un angular-ui router and need to be replaced in your code

**Comment by BriceN** [comment]
@revolunet @Sparksx I agree that this should be fixed in the library, and not with a workaround.

**Comment by revolunet** [comment]
so someone should issue a pull request please, i dont use angular anymore sorry

@justinsa is now the main contributor

**Comment by justinsa** [comment]
My opinion is that the library should stop having an opinion on the route provider used and let the implementer handle that decision. We could then provide a Cookbook as part of the Wiki or the README to explain how to configure different route providers for auto-page tracking or this setPageEvent issue. This library should do Google Analytics the best it can and not need updating due to ng-router or ui-router changes.

If someone wants to issue a PR to fix this issue, great, if not, I will likely make a major version change to alleviate this possibility from being able to happen.
